metalearn/metafeatures/base.py

```diff
@@ -0,0 +1,176 @@
+import inspect
+from abc import ABC, abstractmethod
+from typing import List, Callable, Dict, Union, Optional, Any
+import itertools
+
+from metalearn.metafeatures.constants import ProblemType, MetafeatureGroup
+
+class ResourceComputer:
+
+    def __init__(
+        self,
+        computer: Callable,
+        returns: List[str],
+        argmap: Optional[Dict[str,Any]] = {}
```

Comment:
Should we just use `**kwargs`?

Reply:
I like your guys' idea of simplifying this part of the code. I think it would be good. To answer your question @macetheace96:
> Should we instead just require that each function use the proper resource name or would this still be required if we did that?

We would still need to address the issue of when a metafeature-computing or resource-computing function has arguments that are not global cached resources, such as the `seed` argument in many function signatures like `get_preprocessed_data`.

@bjschoenfeld I'm curious to hear more about what your vision is concerning the use of `**kwargs` here.

Reply:
Currently, `ResourceComputer` gives the flexibility of allowing either the exact resource name or a local alias for the resource name to be used in a function signature. If a local alias is used, then a mapping to the cached resource it refers to must be supplied in the `argmap` parameter of the `ResourceComputer` constructor. If the exact name is used, then no mapping needs to be supplied in the `argmap` parameter. If only exact names are used, then no `argmap` parameter needs to be supplied.

Reply:
I didn't know that, that completely addresses my issue! I'm still interested in hearing Brandon's kwargs idea though.

Reply:
We will not use kwargs. Semantically, it makes more sense to use argmap as is.